include/caffe/common.hpp

```diff
@@ -70,9 +83,13 @@ namespace cv { class Mat; }
 
 namespace caffe {
 
+#ifdef USE_BOOST
 // We will use the boost shared_ptr instead of the new C++11 one mainly
 // because cuda does not work (at least now) well with C++11 features.
 using boost::shared_ptr;
+#else
+using std::shared_ptr;
+#endif  // USE_BOOST
 
```

Comment:
In common.hpp
```cpp
#ifdef USE_BOOST
 // We will use the boost shared_ptr instead of the new C++11 one mainly
 // because cuda does not work (at least now) well with C++11 features.
using boost::shared_ptr;
#else
using std::shared_ptr;
#endif  // USE_BOOST
```